### PR TITLE
Recategorize config classes

### PR DESCRIPTION
**Pull request by alibektas**

### CODE REVIEW

**Comment by alibektas** [comment]
This made most of the ratoml tests invalid, which I will correct in a future PR.

**Comment by lnicola** [comment]
Nit: [this](https://github.com/rust-lang/rust-analyzer/blob/f454ea877180f0fd8d53390a1d9e6bdd6047b35d/crates/rust-analyzer/src/config.rs#L68) should say `~/.config/rust-analyzer/rust-analyzer.toml` instead of `~/.config/rust-analyzer.toml` ([cf.](https://github.com/rust-lang/rust-analyzer/blob/f454ea877180f0fd8d53390a1d9e6bdd6047b35d/crates/rust-analyzer/src/config.rs#L809)).

File: crates/rust-analyzer/src/handlers/notification.rs [hunk 1]
```diff
@@ -182,15 +189,26 @@ pub(crate) fn handle_did_save_text_document(
...
+
+    if !deps_change_processed
+        && state.config.script_rebuild_on_save(None)
+        && state.build_deps_changed
+    {
+        state.build_deps_changed = false;
+        state
+            .fetch_build_data_queue
+            .request_op("build_deps_changed - save notification".to_owned(), ());
+    }
```
Comment (Veykril) [inline crates/rust-analyzer/src/handlers/notification.rs:210]: No need for this, if we there is no vfs path for the document it wont be a source file anyways

File: crates/rust-analyzer/src/config.rs [hunk 2]
```diff
@@ -513,219 +484,243 @@ config_data! {
...
-        procMacro_enable: bool                     = true,
-        /// Internal config, path to proc-macro server executable.
-        procMacro_server: Option<Utf8PathBuf>          = None,
+config_data! {
+    workspace: struct WorkspaceDefaultConfigData <- WorkspaceConfigInput -> {
+        /// Pass `--all-targets` to cargo invocation.
+        cargo_allTargets: bool           = true,
+        /// Automatically refresh project info via `cargo metadata` on
+        /// `Cargo.toml` or `.cargo/config.toml` changes.
+        pub(crate) cargo_autoreload: bool           = true,
```
Comment (Veykril) [inline crates/rust-analyzer/src/config.rs:535]:
```suggestion
        cargo_autoreload: bool           = true,
```
nit this should have an access method instead

**Review by Veykril — APPROVED** [review]

**Pull request merged**